contracts/BaseWrapper.sol

```diff
@@ -197,16 +213,23 @@ abstract contract BaseWrapper {
     ) internal returns (uint256 migrated) {
         VaultAPI _bestVault = bestVault();
 
-        uint256 alreadyDeposited = _bestVault.balanceOf(account).mul(_bestVault.pricePerShare()).div(10**uint256(_bestVault.decimals()));
-        uint256 amountToMigrate = amount.sub(alreadyDeposited);
+        // NOTE: Only override if we aren't migrating everything
+        uint256 _depositLimit = _bestVault.depositLimit();
+        uint256 _totalAssets = _bestVault.totalAssets();
+        if (_depositLimit <= _totalAssets) return 0; // Nothing to migrate (not a failure)
 
-        uint256 depositLeft = _bestVault.depositLimit().sub(_bestVault.totalAssets());
-        if (amountToMigrate > depositLeft) amountToMigrate = depositLeft;
+        uint256 _amount = amount;
+        if (_amount < uint256(-1) && _depositLimit < uint256(-1)) {
```

Comment:
small gas improvement if we change the predicate to:
`(_depositLimit < MAX_UINT && _amount < MAX_UINT )`

Since we are eventually going to uncap the deposit limits in most vaults, this will short circuit and avoid the if block